extensions/web-dependency-locator/runtime/src/main/java/io/quarkus/webdependency/locator/runtime/WebDependencyLocatorRecorder.java

```diff
@@ -11,30 +13,38 @@
 @Recorder
 public class WebDependencyLocatorRecorder {
 
+    private static final Logger LOG = Logger.getLogger(WebDependencyLocatorRecorder.class.getName());
+
     public Handler<RoutingContext> getHandler(String webDependenciesRootUrl,
             Map<String, String> webDependencyNameToVersionMap) {
         return (event) -> {
             String path = event.normalizedPath();
             if (path.startsWith(webDependenciesRootUrl)) {
-                String rest = path.substring(webDependenciesRootUrl.length());
-                String webdep = rest.substring(0, rest.indexOf('/'));
-                if (webDependencyNameToVersionMap.containsKey(webdep)) {
-                    // Check this is not the actual path (ex: /webjars/jquery/${jquery.version}/...
-                    int endOfVersion = rest.indexOf('/', rest.indexOf('/') + 1);
-                    if (endOfVersion == -1) {
-                        endOfVersion = rest.length();
-                    }
-                    String nextPathEntry = rest.substring(rest.indexOf('/') + 1, endOfVersion);
-                    if (webDependencyNameToVersionMap.get(webdep) == null
-                            || nextPathEntry.equals(webDependencyNameToVersionMap.get(webdep))) {
-                        // go to the next handler (which should be the static resource handler, if one exists)
-                        event.next();
+                try {
```

Comment:
Not exactly excited to use a try/catch here but I suppose it's hard to have a precondition?

Reply:
Yea, the idea is to handle any issue, and allow it to go to the next handler, rather than bombing.